public/app/features/alerting/unified/hooks/useCombinedRuleNamespaces.ts

```diff
@@ -494,6 +494,9 @@ export function useCombinedRules(
   result?: CombinedRuleNamespace[];
   error?: unknown;
 } {
+  //  For legacy architecture reasons (DashboardModel) the isNew also respects UID being null
+  const isNewDashboard = dashboardUID === '' || dashboardUID === null;
```

Comment:
not familiar with the changes to the dashboard model, so I assume `dashboardUID === ''` is the new way to check for unsaved dashboards. In that case code change here makes sense!

Reply:
was always like that, it's just the old DashboardModel that changed the empty string to `null` on creation